packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -126,7 +127,7 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
       return;
     }
 
-    if (focusedKey !== lastFocusedKey.current) {
+    if (focusedKey !== lastFocusedKey.current && getInteractionModality() !== 'pointer') {
```

Comment:
This will only apply to components that use virtualizer. Menu will bring items into view on hover [here](https://github.com/adobe/react-spectrum/blob/48baaa114d22347bc24b6eaf1c85afb26f8aba01/packages/%40react-aria/menu/src/useMenuItem.ts#L161-L166)